pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -266,6 +304,77 @@ func (c *podIPMetricsCollector) fillBuckets(diff uint64) {
 	}
 }
 
+// countIPsFronRange returns the number of available hosts in a subnet.
+// The max number is limited by the size of an uint64.
+// Number of hosts is calculated with the formula:
+// IPv4: 2^x – 2, not consider network and broadcast address
+// IPv6: 2^x - 1, not consider network address
+// where x is the number of host bits in the subnet.
+func (c *podIPMetricsCollector) countIPsFromRange(subnet *net.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits <= ones {
+		return 0, fmt.Errorf("invalid subnet mask: %v", subnet.Mask)
+	}
+	// this checks that we are not overflowing an int64
+	if bits-ones >= 64 {
+		return math.MaxUint64, nil
+	}
+	max := uint64(1) << uint(bits-ones)
+	max--
+	if subnet.IP.To4() != nil {
+		// Don't use the IPv4 network's broadcast address
+		if max == 0 {
+			return 0, fmt.Errorf("subnet includes only the network and broadcast addresses")
+		}
+		max--
+	} else if max == 0 {
+		return 0, fmt.Errorf("subnet includes only the network address")
+	}
+	return max, nil
+}
+
+func (c *podIPMetricsCollector) updateAssignedIPs(subnet *net.IPNet, totalIP uint64) {
+	if subnet.IP.To16() != nil && subnet.IP.To4() == nil {
+		c.assignedIPv6AddrCount += totalIP
+	} else if subnet.IP.To4() != nil {
+		c.assignedIPv4AddrCount += totalIP
+	}
+}
+
+func (c *podIPMetricsCollector) calculateAssignedIP() error {
+	node, err := c.clientset.CoreV1().Nodes().Get(context.Background(), c.nodeName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error getting node %s: %v", c.nodeName, err)
+	}
+
+	podCIDRs := node.Spec.PodCIDRs
+	if len(podCIDRs) == 0 && node.Spec.PodCIDR != "" {
+		podCIDRs = []string{node.Spec.PodCIDR}
+	}
+	if len(podCIDRs) == 0 {
+		return fmt.Errorf("both podCIDR and podCIDRs are empty")
+	}
```

Comment:
Ah you're checking this too. Maybe this for one less evaluation?

```
	podCIDRs := node.Spec.PodCIDRs
	if len(podCIDRs) == 0 {
		if node.Spec.PodCIDR == "" {
			return fmt.Errorf("both podCIDR and podCIDRs are empty")
		} else {
			podCIDRs = []string{node.Spec.PodCIDR}
		}
	}
```

Reply:
Done.